lib/Dialect/StaticLogic/StaticLogicOps.cpp

```diff
@@ -180,6 +183,59 @@ void PipelineWhileOp::build(OpBuilder &builder, OperationState &state,
                                        ValueRange());
 }
 
+/// Get the terminator of the pipeline.
+PipelineTerminatorOp PipelineWhileOp::getTerminator() {
+  return cast<PipelineTerminatorOp>(getStagesBlock().getTerminator());
+}
+
+/// Compute the distance between the two stages.
+uint64_t PipelineWhileOp::computeDistance(PipelineStageOp from,
```

Comment:
Is it possible to implement this function in a safer way? Specifically, in the (incorrect) case when if `to` is earlier in the pipeline than `from`. how about using `from->getIterator()/to->getIterator()` and comparing those?

Reply:
Good question, I haven't tried that but I will check.

Reply:
I couldn't figure out how to do something fancy with the iterators, they appear to only support equality comparisons and increments. I will at least add a check `from->isBeforeInBlock(to)` to assert this loop will terminate.

Reply:
Hmm, locally, `std::distance(op1->getIterator(), op2->getIterator())` seems to compile.

Reply:
oh i see, i will use std::distance.